src/templates/filter-bar-controller.js

```diff
@@ -33,7 +39,20 @@ FilterBarController.prototype.getFilteredNotes = function(notes) {
     }
 
     const filterFunction = (this.selected && this.selected.filter) || (a => a);
-    return notes.filter(filterFunction);
+    let filteredNotes = notes.filter(filterFunction);
+
+    // Prevent notes in the unread filter from disappearing when marked as read.
+    if (this.selected && this.selected.name === 'unread') {
+        const noteReads = store.getState().notes.noteReads;
+
+        const unreadsFilter = note => (note.id in noteReads && noteReads[note.id] === 'unread');
+        filteredNotes = notes.filter(unreadsFilter)
+            .concat(filteredNotes)
+            .sort((note1, note2) => (new Date(note2.timestamp) - new Date(note1.timestamp))
+        );
```

Comment:
Ahh nice, you're right. I updated it to a better version in 7a3ac5f. Although I think you might have some ideas on how to do that with fewer lines of code ;)